pkg/thing/interactors/publish_data.go

```diff
@@ -0,0 +1,93 @@
+package interactors
+
+import (
+	"fmt"
+
+	"github.com/CESARBR/knot-babeltower/pkg/network"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+)
+
+var (
+	// ErrNoAuthToken error occur when no authorization token is provided
+	ErrNoAuthToken = fmt.Errorf("missing authorization token")
+
+	// ErrNoIDParam error occur when no thing's id is provided
+	ErrNoIDParam = fmt.Errorf("missing thing's id")
+
+	// ErrNoDataParam error occur when no thing's data is provided
+	ErrNoDataParam = fmt.Errorf("missing thing's data")
+
+	// ErrNoSchema error occur when the thing has no schema yet
+	ErrNoSchema = fmt.Errorf("thing has no schema")
+
+	// ErrDataInvalid error occur when the provided data mismatch the thing's schema
+	ErrDataInvalid = fmt.Errorf("data is incompatible with thing's schema")
+)
+
+// PublishData executes the use case operations to publish data from the things to cloud
+func (i *ThingInteractor) PublishData(authorization, thingID string, data []network.Data) error {
+	err := i.verifyThingData(authorization, thingID, data)
+	if err != nil {
+		return fmt.Errorf("unable to update data: %w", err)
+	}
+
+	// TODO: publishing data to local storage
```

Comment:
I think this comment could be removed because further the `babeltower` won't know to whom it's publishing the data.

Reply:
Done!